src/utils/buildBalanceDisplays.ts

```diff
@@ -3,45 +3,55 @@ import { translateRaw } from '@translations';
 
 import { weiToFloat, convertToFiatFromAsset } from './convert';
 
+const buildAccountDisplayBalances = (
+  accounts: StoreAccount[],
+  asset: StoreAsset,
+  exchangeRate: number | undefined
+) =>
+  accounts.reduce((acc, currAccount) => {
+    const matchingAccAssets = currAccount.assets.filter((accAsset) => accAsset.uuid === asset.uuid);
+    if (matchingAccAssets.length) {
+      return [
+        ...acc,
+        ...matchingAccAssets.map((accAsset) => ({
+          address: currAccount.address,
+          ticker: accAsset.ticker,
+          amount: weiToFloat(accAsset.balance, accAsset.decimal),
+          fiatValue: convertToFiatFromAsset(accAsset, exchangeRate),
+          label: currAccount.label
+        }))
+      ];
+    }
+    return acc;
+  }, [] as BalanceAccount[]);
+
+const buildBalance = (
+  accounts: StoreAccount[],
+  getAssetRate: (asset: Asset) => number | undefined
+) => (asset: StoreAsset) => {
+  const exchangeRate = getAssetRate(asset);
+  return {
+    id: `${asset.name}-${asset.ticker}`,
+    name: asset.name || translateRaw('WALLET_BREAKDOWN_UNKNOWN'),
+    ticker: asset.ticker,
+    uuid: asset.uuid,
+    amount: weiToFloat(asset.balance, asset.decimal),
+    fiatValue: convertToFiatFromAsset(asset, exchangeRate),
```

Comment:
We should never store a float in memory.
Preserve a BN, or if it's only a view helper, store a string.